### ISSUE TITLE
Blocks rendered too many time even if not used in front

### ISSUE THREAD

**Issue by Hydrog3n** [PrestaSafe/prettyblocks#283]
Hello,
We are using prettyblocks and prettyblockspro. 

We notice than using blocks (we create some custom for us) we notice [here](https://github.com/PrestaSafe/prettyblocks/blob/main/classes/PrettyBlocksModel.php#L154-L162) we will loadBlock but that will request all [available blocks ](https://github.com/PrestaSafe/prettyblocks/blob/main/classes/PrettyBlocksModel.php#L80C25-L80C43)

The pb is this method will render all blocks and the ProductGridMansonry use a default function : `HelperBuilder::getRandomProduct()` that use a ORDER BY RAND() and this on 33K products it's a lot for our DB

On our case that function will be calls for 8 blocks on the zone 16 times due to that part of [the code](https://github.com/PrestaSafe/prettyblocks/blob/main/classes/PrettyBlocksModel.php#L156-L159) doing `mergeStateWithFields` twice. You multiple this by the traffic we successly overloaded the CPU of the Db

I can do a PR to put in cache the `getBlocksAvailable` results return but I'm wondering if we could think to something better. 

Today because we don't using blocks from the pro modules we remove the attachement of the module to `ActionRegisterBlock`

**Comment by Hydrog3n** [comment]
hi, 
Clearly that the simple to removed that but this will not removed the risk and will not improved the preformance. 

The solution you want on our side we already remove on the code.

**Comment by PrestaSafe** [comment]
Ok great. If you can make a PR, it's could be very nice. 
I will a lot of improvents end of 2024 on Prettyblocks. 
I put it on my todo list if you do not make this PR. 
Thanks !

**Comment by Hydrog3n** [comment]
Yes I will try but maybe not before january with same lot of projects ! 
Let know here if you start to do it before me ;) 

Thank you

**Comment by Hydrog3n** [comment]
I propose the PR let me know if that follow your guideline and suits you :)
